hw/ip/edn/doc/theory_of_operation.md

```diff
@@ -28,19 +30,25 @@ There are two broad modes for state machine control: auto request mode and boot-
 Random values are needed by peripherals almost immediately after reset, so to simplify interactions with the boot ROM, boot-time request mode is the default mode.
 
 In boot-time request mode, the command sequence is fully hardware-controlled and no command customization is possible.
-In this mode, the EDN automatically issues a special reduced-latency `instantiate` command followed by the default `generate` commands.
+In this mode, EDN automatically issues a special reduced-latency `instantiate` command followed by the default `generate` commands.
 This means, for instance, that no personalization strings or additional data may be passed to the CSRNG application interface port in this mode.
-On exiting, the EDN issues an `uninstantiate` command to destroy the associated CSRNG instance.
 
 Once firmware initialization is complete, it is important to exit this mode if the endpoints ever need FIPS-approved random values.
-Should another generate command be needed, it can only be issued after exiting boot mode.
-This is done by either *clearing* the `EDN_ENABLE` field or *clearing* the `BOOT_REQ_MODE` field in [`CTRL`](registers.md#ctrl) to halt the boot-time request state machine.
-Firmware must then wait for the transition of the state machine by polling the `CMD_RDY` field of the [`SW_CMD_STS`](registers.md#sw_cmd_sts) register or wait for the state machine to enter the SW mode by polling the [`MAIN_SM_STATE`](registers.md#main_sm_state) register.
+Should another generate command be needed, it can only be issued after exiting boot-time request mode.
+
+To exit boot-time request mode, firmware has two options:
```

Comment:
Nitty grammar point: This is correct, but it probably reads more smoothly the other way around. "There are two ways that firmware can exit boot-time request mode:"